guardian/support-service-lambdas
# Zuora <-> salesforce link remover - update billing account in zuora

## What does this change?
Create a new lambda that take in a billing account Id, updates the billing account crmId in Zuora to null, and returns the status of the update in the output from the lambda

<img width="400" alt="Screenshot 2024-07-01 at 15 52 54" src="https://github.com/guardian/support-service-lambdas/assets/36296660/42ae1a60-7291-4a4b-b7ea-c09b885bb19c">

<img width="563" alt="Screenshot 2024-07-01 at 14 08 29" src="https://github.com/guardian/support-service-lambdas/assets/36296660/d245942f-cf70-49c7-bab7-887201a29d56">

## How to test

**Testing a successful update**

- Run the lambda with the following input:

```
{
		attributes: {
			type: 'Zuora__CustomerAccount__c',
			url: '/services/data/v54.0/sobjects/Zuora__CustomerAccount__c/a029E00000OEcxWQAT',
		},
		Id: 'a029E00000OEcxWQAT',
		Zuora__Account__c: '0019E00001JW6SLQA1',
		GDPR_Removal_Attempts__c: 0,
		Zuora__External_Id__c: '2c92c0f875e014d30175e5a18c51068b',
	}
```

- verify that billing account with Id 2c92c0f875e014d30175e5a18c51068b in Zuora has had its crmId updated to null
- verify the output of the lambda is:
![image](https://github.com/guardian/support-service-lambdas/assets/36296660/4cb011f2-c273-4256-9e62-54f00accf78d)

**Testing a failed update**
- update the billing account id in the input to one that doesn't exist in zuora, i.e, use the following input.
```
{
		attributes: {
			type: 'Zuora__CustomerAccount__c',
			url: '/services/data/v54.0/sobjects/Zuora__CustomerAccount__c/a029E00000OEcxWQAT',
		},
		Id: 'a029E00000OEcxWQAT',
		Zuora__Account__c: '0019E00001JW6SLQA1',
		GDPR_Removal_Attempts__c: 0,
		Zuora__External_Id__c: 'abc123', //dodgy billing account Id
	}
```
- Run the lambda
- verify the output of the lambda is:
![image](https://github.com/guardian/support-service-lambdas/assets/36296660/27369d81-774f-4a3e-9cba-43f4a3a1fea5)
